
avr-ibc-1.8.0 seems to have depreciated some of the required features
by the Arduino libraries, resulting in errors:

  hardware/arduino/cores/arduino/HardwareSerial.cpp:91:41:
      error: attempt to use poisoned "SIG_USART0_RECV"

This patch to the Arduino sources works around that issue.

See also:
  http://arduino.cc/forum/index.php?topic=92364.0


diff --git a/hardware/arduino/cores/arduino/Arduino.h b/hardware/arduino/cores/arduino/Arduino.h
index 830c995..8b378fc 100755
--- a/hardware/arduino/cores/arduino/Arduino.h
+++ b/hardware/arduino/cores/arduino/Arduino.h
@@ -5,6 +5,7 @@
 #include <string.h>
 #include <math.h>
 
+#define __AVR_LIBC_DEPRECATED_ENABLE__ 1
 #include <avr/pgmspace.h>
 #include <avr/io.h>
 #include <avr/interrupt.h>
